addons/t9n/models/resource.py

```diff
@@ -14,3 +19,52 @@ class Resource(models.Model):
     project_id = fields.Many2one(
         comodel_name="t9n.project",
     )
+
+    _sql_constraints = [
+        (
+            "file_name_unique",
+            "unique(file_name)",
+            "A file with the same name already exists!"
+        ),
+    ]
+
+    def _decode_resource_file(self, resource_file):
+        try:
+            file_content = base64.b64decode(resource_file).decode()
+            po_obj = polib.pofile(file_content)
+        except (IOError, UnicodeDecodeError):
+            raise ValidationError(_("The file should be a .po file with a valid syntax."))
```

Comment:
Would be great to indicate which file(s) failed to upload in case the user is uploading multiple files at once

Reply:
Done!